src/Picasso_ParticleInit.hpp

```diff
@@ -368,6 +369,142 @@ void initializeParticles( InitUniform, const ExecutionSpace& exec_space,
+
+  \tparam ParticleListType The type of particle list to initialize.
+
+  \tparam FacetGeometry The type of the facet geometry representing the
+  surface to initialize with particles
+
+  \tparam InitFunctor Initialization functor type. See the documentation below
+  for the create_functor parameter on the signature of this functor.
+
+  \param Initialization type tag.
+
+  \param particles_per_facet The number of particles to populate each cell
+  dimension with.
+
+  \param surface A FacetGeometry type (FacetGeoemtry or MarchingCubes data)
+  containing the list of surface facets to seed the particle list
+
+  \param create_functor A functor which populates a particle given the logical
+  position of a particle. This functor returns true if a particle was created
+  and false if it was not giving the signature:
+
+      bool createFunctor( const double px[3],
+                          typename ParticleAoSoA::tuple_type& particle );
+
+  \param particle_list The list of particles to populate. This will be filled
+  with particles and resized to a size equal to the number of particles
+  created.
+*/
+template <class ParticleListType, class InitFunc, class FacetGeometry,
+          class ExecutionSpace>
+void initializeParticlesSurface( InitRandom, const ExecutionSpace& exec_space,
+                                 const int particles_per_facet,
+                                 const FacetGeometry& surface,
+                                 const InitFunc& create_functor,
+                                 ParticleListType& particle_list )
+{
+    // Memory space.
+    using memory_space = typename ParticleListType::memory_space;
+
+    // Particle type.
+    using particle_type = typename ParticleListType::particle_type;
+
+    // Get the local grid.
+    const auto& local_grid = *( particle_list.mesh().localGrid() );
+
+    // Create a local mesh.
+    auto local_mesh = Cajita::createLocalMesh<ExecutionSpace>( local_grid );
+
+    // Get the global grid.
+    const auto& global_grid = local_grid.globalGrid();
+
+    // Get the particles.
+    auto& particles = particle_list.aosoa();
+
+    // Allocate enough space for the case the particles consume the entire
+    // local grid.
+    int num_facets = surface.facets.extent( 0 );
+    int num_particles = particles_per_facet * num_facets;
+    particles.resize( num_particles );
+
+    // Creation status.
+    auto particle_created = Kokkos::View<bool*, memory_space>(
+        Kokkos::ViewAllocateWithoutInitializing( "particle_created" ),
+        num_particles );
+
+    // Create a random number generator.
+    uint64_t seed =
+        global_grid.blockId() + ( 19383747 % ( global_grid.blockId() + 1 ) );
+    using rnd_type = Kokkos::Random_XorShift64_Pool<ExecutionSpace>;
+    rnd_type pool;
+    pool.init( seed, num_facets );
+
+    // // Initialize particles.
+    Kokkos::parallel_for(
+        "Picasso::ParticleInit::Uniform",
+        Kokkos::RangePolicy<ExecutionSpace>( 0, num_facets ),
```

Comment:
From discussion: fixme comment for iterating over `num_particles` instead

Reply:
Done.